psi4/src/psi4/libmints/twobody.cc

```diff
@@ -56,24 +60,277 @@ TwoBodyAOInt::TwoBodyAOInt(const IntegralFactory *intsfactory, int deriv)
       source_full_(nullptr),
       source_(nullptr),
       deriv_(deriv) {
-    // outfile->Printf( "TwoBodyAOInt object created with: %s, %s, %s, %s\n",
-    //        original_bs1_->name().c_str(),
-    //        original_bs2_->name().c_str(),
-    //        original_bs3_->name().c_str(),
-    //        original_bs4_->name().c_str());
     // The derived classes allocate this memory.
-    force_cartesian_ = false;
     tformbuf_ = nullptr;
     natom_ = original_bs1_->molecule()->natom();  // This assumes the 4 bases come from the same molecule.
+
+    // Figure out how equivalent
+    bra_same_ = (original_bs1_ == original_bs2_);
+    ket_same_ = (original_bs3_ == original_bs4_);
+    braket_same_ = (original_bs1_ == original_bs3_ && original_bs2_ == original_bs4_);
+
+    // Setup sieve data
+    screening_threshold_ = Process::environment.options.get_double("INTS_TOLERANCE");
+    auto screentype = Process::environment.options.get_str("SCREENING");
+    if (screentype == "SCHWARZ")
+        screening_type_ = ScreeningType::Schwarz;
+    else if (screentype == "CSAM")
+        screening_type_ = ScreeningType::CSAM;
+    else
+        throw PSIEXCEPTION("Unknown screening type " + screentype + " in TwoBodyAOInt()");
+    
+    if (screening_threshold_ == 0.0) screening_type_ = ScreeningType::None;
```

Comment:
I'm not sure how much of a problem this is. I think we are seeking to test for the repr of float 0. And l2 is the realm where 1.e-34 vs 1.e-150 can mean something.

Reply:
Not sure either. Allowing super small numbers for the screening value might make it difficult working with epsilon functions. A safer option is checking for negative numbers (to my believe) instead of equal zero. If you or Andy  think it's fine as it is then it's ok for me, too.